src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,86 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-/**
- * Make this class immutable. See requirements in the README file
- */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car implements Cloneable {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.wheels = getClonedList(wheels);
+        this.engine = engine == null ? null : engine.clone();
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    private List<Wheel> getClonedList(List<Wheel> wheels) {
+        List<Wheel> newList = new ArrayList<>();
```

Comment:
```suggestion
        List<Wheel> newList = new ArrayList<>(wheels.size());
```
if you know the capacity will be more than 10, it's good to set it from the beginning. performance wise